packages/twenty-front/src/modules/activities/emails/components/EmailThreads.tsx

```diff
@@ -39,42 +51,87 @@ export const EmailThreads = ({
   entity: ActivityTargetableObject;
 }) => {
   const { openEmailThread } = useEmailThread();
-
   const { enqueueSnackBar } = useSnackBar();
 
-  const threadQuery =
+  const [emailThreadsPage, setEmailThreadsPage] =
+    useRecoilState<EmailThreadsPageType>(emailThreadsPageState);
+
+  const [isFetchingMoreEmails, setIsFetchingMoreEmails] = useState(false);
+
+  const [threadQuery, queryName] =
     entity.targetObjectNameSingular === CoreObjectNameSingular.Person
-      ? getTimelineThreadsFromPersonId
-      : getTimelineThreadsFromCompanyId;
-
-  const threadQueryVariables = {
-    ...(entity.targetObjectNameSingular === CoreObjectNameSingular.Person
-      ? { personId: entity.id }
-      : { companyId: entity.id }),
-    page: 1,
-    pageSize: TIMELINE_THREADS_DEFAULT_PAGE_SIZE,
-  };
-
-  const threads = useQuery(threadQuery, {
+      ? [getTimelineThreadsFromPersonId, 'getTimelineThreadsFromPersonId']
+      : [getTimelineThreadsFromCompanyId, 'getTimelineThreadsFromCompanyId'];
+
+  const threadQueryVariables = useMemo(
+    () => ({
+      ...(entity.targetObjectNameSingular === CoreObjectNameSingular.Person
+        ? { personId: entity.id }
+        : { companyId: entity.id }),
+      page: 1,
+      pageSize: TIMELINE_THREADS_DEFAULT_PAGE_SIZE,
+    }),
+    [entity],
+  ) as GetTimelineThreadsFromPersonIdQueryVariables;
+
+  const { data, loading, fetchMore, error } = useQuery(threadQuery, {
```

Comment:
We try to not use too much useMemo and useCallback when they are not necessarily needed. Here it seems that they are not mandatory.